src/main/java/org/apache/commons/lang3/ClassUtils.java

```diff
@@ -458,52 +458,90 @@ public static String getCanonicalName(final Object object, final String valueIfN
     }
 
     /**
-     * Converts a given name of class into canonical format. If name of class is not a name of array class it returns
-     * unchanged name.
-     *
+     * Converts a given internal class name into its canonical format.
      * <p>
-     * The method does not change the {@code $} separators in case the class is inner class.
+     * If the name is a valid Java array type descriptor (e.g., "[I", "[[Ljava.lang.String;"),
+     * it will be converted to a canonical form such as "int[]" or "java.lang.String[][]".
+     * If the name is a non-array class name (e.g., "java.lang.String"), it will be returned as-is
+     * after whitespace trimming.
      * </p>
      *
-     * <p>
-     * Example:
+     * <p>Special handling includes:
+     * <ul>
+     *   <li>{@code null} or blank input returns an empty string</li>
+     *   <li>Invalid or malformed input will result in an {@link IllegalArgumentException}</li>
+     * </ul>
+     * </p>
+     *
+     * <p>Examples:
      * <ul>
-     * <li>{@code getCanonicalName("[I") = "int[]"}</li>
-     * <li>{@code getCanonicalName("[Ljava.lang.String;") = "java.lang.String[]"}</li>
-     * <li>{@code getCanonicalName("java.lang.String") = "java.lang.String"}</li>
+     *   <li>{@code getCanonicalName("[I")} → {@code "int[]"}</li>
+     *   <li>{@code getCanonicalName("[[Ljava.lang.String;")} → {@code "java.lang.String[][]"}</li>
+     *   <li>{@code getCanonicalName("java.lang.String")} → {@code "java.lang.String"}</li>
      * </ul>
      * </p>
      *
-     * @param name the name of class.
-     * @return canonical form of class name.
+     * @param name the internal name of a class
+     * @return canonical form of the class name, or an empty string if input is blank or null
+     * @throws IllegalArgumentException if the class name is invalid or malformed
      */
+
     private static String getCanonicalName(final String name) {
+        if (StringUtils.isBlank(name)) {
```

Comment:
This doesn't seem right. How is "    " a valid class name? Or null for that matter...